recsyslearn/segmentations.py

```diff
@@ -1,140 +1,186 @@
+
+
+        Parameters
+        ----------
+        dataset : pd.DataFrame
+            The complete dataset.
+
+        proportion : list, default [0.8, 0.2]
+            The proportion of interactions wanted for every group. Its length should be between 1 and 3.
+
+
+        Raises
+        ------
+        SegmentationNotSupportedException
+            If len(proportion) not in (1, 2, 3).
+
+        WrongProportionsException
+            If sum(proportion) is not 1, which means it doesn't cover all the items/users.
+
 
-	'''
-	Segmentation of items based on the number of interaction they have.
-	'''
+        Return
+        ------
+        DataFrame with items and belonging group.
+        """
 
-	def segment(self, dataset: pd.DataFrame, proportion: list = [0.8, 0.2], min_interaction: int = 0) -> pd.DataFrame:
+        if len(proportion) not in (1, 2, 3):
+            raise SegmentationNotSupportedException()
 
-		'''
-		Segmentation of items based on their interactions with different users.
+        if sum(proportion) != 1:
+            raise WrongProportionsException()
 
+        if len(proportion) == 1:
+            return dataset
 
-		Parameters
-		----------
-		dataset : pd.DataFrame
-			The complete dataset.
+        item_groups = dataset.groupby('item').size().reset_index(name='count').sort_values('count', ascending=False)
 
-		proportion : list, default [0.8, 0.2]
-			The proportion of interactions wanted for every group. Its length should be between 1 and 3.
+        tmp = item_groups[item_groups['count'] > min_interaction]
+        n_int = tmp['count'].sum()
 
+        short_thr = np.rint(proportion[0] * n_int)
+        mid_thr = np.rint(proportion[1] * n_int) + short_thr
+        tmp.loc[:, 'cumulative_sum'] = tmp['count'].cumsum()
 
-		Raises
-		------
-		SegmentationNotSupportedException
-			If len(proportion) not in (1, 2, 3).
-		
-		WrongProportionsException
-			If sum(proportion) is not 1, which means it doesn't cover all the items/users.
+        short_head = tmp.loc[tmp['cumulative_sum'].lt(short_thr), 'item']
+        mid_tail = tmp.loc[tmp['cumulative_sum'].lt(mid_thr) & ~tmp['item'].isin(short_head), 'item']
+        conditions = [item_groups['item'].isin(short_head), item_groups['item'].isin(mid_tail)]
+        choices = ['1', '2']
+        item_groups.loc[:, 'group'] = np.select(conditions, choices, default='3' if len(proportion) == 3 else '2')
+        return item_groups[['item', 'group']]
 
-		
-		Return
-		------
-		DataFrame with items and belonging group.
-		'''
 
-		if len(proportion) not in (1, 2, 3):
-			raise SegmentationNotSupportedException()
+class PopularityPercentage(Segmentation):
+    """
+    Calculate item or user popularity based on the percentage of interaction they have.
+    """
 
-		if sum(proportion) != 1:
-			raise WrongProportionsException()
+    def segment(self, dataset: pd.DataFrame, calculate_on: str = 'item') -> pd.DataFrame:
```

Comment:
To be consistent with what we have done with other functions, I suggest to use `group` instead of `calculate_on`